core/templates/domain/question/editable-question-backend-api.service.ts

```diff
@@ -41,13 +41,30 @@ angular.module('oppia').factory('EditableQuestionBackendApiService', [
       QUESTION_SKILL_LINK_URL_TEMPLATE) {
     var _createQuestion = function(
         skillIds, skillDifficulties,
-        questionDict, successCallback, errorCallback) {
+        questionDict, imagesData, successCallback, errorCallback) {
       var postData = {
         question_dict: questionDict,
         skill_ids: skillIds,
         skill_difficulties: skillDifficulties
       };
-      $http.post(QUESTION_CREATION_URL, postData).then(function(response) {
+      let body = new FormData();
+      body.append('payload', JSON.stringify(postData));
+      let filenames = imagesData.map(obj => obj.filename);
+      let imageBlobs = imagesData.map(obj => obj.imageBlob);
+      for (let idx in imageBlobs) {
+        body.append(filenames[idx], imageBlobs[idx]);
+      }
+      $http.post(QUESTION_CREATION_URL, (body), {
+        // The actual header to be added for form-data is 'multipart/form-data',
+        // But adding it manually won't work because we will miss the boundary
+        // parameter. When we keep 'Content-Type' as undefined the browser
+        // automatically fills the boundary parameter according to the form
+        // data. Refer https://stackoverflow.com/questions/37039852/. and
+        // https://stackoverflow.com/questions/34983071/.
+        headers: {
+          'Content-Type': undefined
```

Comment:
is this how we have handled this for images in skill?

Reply:
No for skills, this is done in angular 8 using http. We don't pass the headers for saving images in skills(I'm not sure why).
@aks681 any suggestions for this?

Reply:
Hmm, ok, in that case, maybe add in the comment that when this is converted to Angular 8, this should be removed and should follow the convention in SkillCreationBackendApiService?

Reply:
done